tests/test_modeling_flax_common.py

```diff
@@ -826,6 +826,86 @@ def _check_attentions_validity(attentions):
             else:
                 _check_attentions_validity(outputs.attentions)
 
+    def test__do_init(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        config.return_dict = True
+
+        for model_class in self.all_model_classes:
+            model = model_class(config, _do_init=False)
+
+            # Check that accesing parmas raises an ValueError when _do_init is False
+            with self.assertRaises(ValueError):
+                params = model.params
+
+            # Check if we params can be properly initialized when calling init_weights
+            params = model.init_weights(model.key, model.input_shape)
+            self.assertIsInstance(params, FrozenDict)
+            # Check if all required parmas are initialized
+            keys = set(flatten_dict(unfreeze(params)).keys())
+            self.assertTrue(all(k in keys for k in model.required_params))
+            # Check if the shapes match
+            flat_params = flatten_dict(unfreeze(params))
+            for k, v in flatten_dict(unfreeze(model.params_shape_tree)).items():
+                self.assertEqual(
+                    v.shape,
+                    flat_params[k].shape,
+                    "Shapes of {} do not match. Expecting {}, got {}.".format(k, v.shape, flat_params[k].shape),
+                )
+
+            # Check that setting params raises an ValueError when _do_init is False
+            with self.assertRaises(ValueError):
+                model.params = params
+
+            # Check if we can do a forward pass
+            inputs_dict["output_hidden_states"] = True
+            inputs = self._prepare_for_class(inputs_dict, model_class).copy()
+            model(**inputs, params=params)
+
+    def test_from_pretrained_with__do_init(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
```

Comment:
IMO we also need to test that the init function can initialize all weights that have not been downloaded or do you want to handle this in a follow up PR? 

E.g. we need to be sure that when doing `from_pretrained(_no_init=True)` that the user is able to do:

```python
params = model.init_weights(params=params)
output = model(input_ids, params=params)
```

Reply:
I think we should make sure this works in this PR (no need to have the fast init here, but at least it should work) to do a forward pass afterwards. Otherwise the method is a bit useless in `from_pretrained(...)` no?